### ISSUE TITLE
Portfolio: USD/JPY net_position

### ISSUE THREAD

**Issue by rsmb7z** [nautechsystems/nautilus_trader#1033]
# Bug Report

A weird issue when opening `USD/JPY` position. As you can see in below logs `OrderFilled` with `100_000` quantity however Portfolio `net_position=99998.0` - I believe it is doing something with commission which is $2 here. This is the case with other similar pairs as well. i.e `USD/CAD`.

```
2023-03-02T09:55:00.676246500Z [INF] TESTER-001.MyTestStrategy-003: <--[EVT] OrderInitialized(instrument_id=USD/JPY.IDEALPRO, client_order_id=O-20230302-001-003-1, side=BUY, type=LIMIT, quantity=100_000.0, time_in_force=GTD, post_only=False, reduce_only=False, options={'price': '136.495', 'display_qty': None, 'expire_time_ns': 1677751200675719200}, emulation_trigger=NO_TRIGGER, contingency_type=OTO, order_list_id=OL-20230302-001-003-1, linked_order_ids=['O-20230302-001-003-2'], parent_order_id=None, tags=None).
2023-03-02T09:55:00.676269500Z [INF] TESTER-001.MyTestStrategy-003: <--[EVT] OrderInitialized(instrument_id=USD/JPY.IDEALPRO, client_order_id=O-20230302-001-003-2, side=SELL, type=TRAILING_STOP_MARKET, quantity=100_000.0, time_in_force=GTC, post_only=False, reduce_only=False, options={'trigger_price': '136.425', 'trigger_type': 'DEFAULT', 'trailing_offset': '0.045', 'trailing_offset_type': 'PRICE', 'expire_time_ns': 0}, emulation_trigger=NO_TRIGGER, contingency_type=NO_CONTINGENCY, order_list_id=OL-20230302-001-003-1, linked_order_ids=None, parent_order_id=O-20230302-001-003-1, tags=None).
2023-03-02T09:55:00.676462700Z [INF] TESTER-001.MyTestStrategy-003: [CMD]--> SubmitOrderList(order_list=OrderList(id=OL-20230302-001-003-1, instrument_id=USD/JPY.IDEALPRO, strategy_id=MyTestStrategy-003, orders=[LimitOrder(BUY 100_000.0 USD/JPY.IDEALPRO LIMIT @ 136.495 GTD 2023-03-02T10:00:00.675Z, status=INITIALIZED, client_order_id=O-20230302-001-003-1, venue_order_id=None, contingency_type=OTO, linked_order_ids=[O-20230302-001-003-2], tags=None), TrailingStopMarketOrder(SELL 100_000.0 USD/JPY.IDEALPRO TRAILING_STOP_MARKET[DEFAULT] @ 136.425-STOP 0.045-TRAILING_OFFSET[PRICE] GTC, status=INITIALIZED, client_order_id=O-20230302-001-003-2, venue_order_id=None, parent_order_id=O-20230302-001-003-1, tags=None)]), position_id=None, exec_algorithm_specs=None).
2023-03-02T09:55:00.682434800Z [INF] TESTER-001.MyTestStrategy-003: <--[EVT] OrderSubmitted(instrument_id=USD/JPY.IDEALPRO, client_order_id=O-20230302-001-003-1, account_id=InteractiveBrokers-DU1234567, ts_event=1677750900681569600).
2023-03-02T09:55:00.682517200Z [INF] TESTER-001.MyTestStrategy-003: <--[EVT] OrderSubmitted(instrument_id=USD/JPY.IDEALPRO, client_order_id=O-20230302-001-003-2, account_id=InteractiveBrokers-DU1234567, ts_event=1677750900681826400).
2023-03-02T09:55:01.173025900Z [INF] TESTER-001.MyTestStrategy-003: <--[EVT] OrderAccepted(instrument_id=USD/JPY.IDEALPRO, client_order_id=O-20230302-001-003-2, venue_order_id=465, account_id=InteractiveBrokers-DU1234567, ts_event=1677750901170946900).
2023-03-02T09:55:01.210348900Z [INF] TESTER-001.MyTestStrategy-003: <--[EVT] OrderAccepted(instrument_id=USD/JPY.IDEALPRO, client_order_id=O-20230302-001-003-1, venue_order_id=464, account_id=InteractiveBrokers-DU1234567, ts_event=1677750901208068000).
2023-03-02T09:55:01.215426600Z [INF] TESTER-001.InteractiveBrokersClient-002: ANSWER position {'account': 'DU1234567', 'contract': 2135092325792: 15016059,USD,CASH,,0,,,,,JPY,USD.JPY,USD.JPY,False,,,,combo:, 'position': Decimal('100000'), 'avg_cost': 136.474}
2023-03-02T09:55:01.216010100Z [INF] TESTER-001.InteractiveBrokersClient-002: ANSWER orderStatus {'order_id': 464, 'status': 'Filled', 'filled': Decimal('100000'), 'remaining': Decimal('0'), 'avg_fill_price': 136.474, 'perm_id': 1096798733, 'parent_id': 0, 'last_fill_price': 136.474, 'client_id': 2, 'why_held': '', 'mkt_cap_price': 0.0}
2023-03-02T09:55:01.216081600Z [INF] TESTER-001.InteractiveBrokersClient-002: ANSWER commissionReport {'commission_report': 2135092944064: ExecId: 0000e215.63ffe227.01.01, Commission: 2.004023, Currency: USD, RealizedPnL: , Yield: , YieldRedemptionDate: 0}
2023-03-02T09:55:01.221692900Z [INF] TESTER-001.MyTestStrategy-003: <--[EVT] OrderFilled(instrument_id=USD/JPY.IDEALPRO, client_order_id=O-20230302-001-003-1, venue_order_id=464, account_id=InteractiveBrokers-DU1234567, trade_id=0000e215.63ffe227.01.01, position_id=USD/JPY.IDEALPRO-MyTestStrategy-003, order_side=BUY, order_type=LIMIT, last_qty=100_000.0, last_px=136.474 JPY, commission=2.00 USD, liquidity_side=NO_LIQUIDITY_SIDE, ts_event=1677750900000000000).
2023-03-02T09:55:01.221757300Z [DBG] TESTER-001.Cache: Indexed PositionId('USD/JPY.IDEALPRO-MyTestStrategy-003'), client_order_id=O-20230302-001-003-1, strategy_id=MyTestStrategy-003).
2023-03-02T09:55:01.221762300Z [DBG] TESTER-001.Cache: Added Position(id=USD/JPY.IDEALPRO-MyTestStrategy-003, strategy_id=MyTestStrategy-003).
2023-03-02T09:55:01.222199700Z [INF] TESTER-001.Portfolio: USD/JPY.IDEALPRO net_position=99998.0
2023-03-02T09:55:01.222222400Z [INF] TESTER-001.MyTestStrategy-003: <--[EVT] PositionOpened(instrument_id=USD/JPY.IDEALPRO, position_id=USD/JPY.IDEALPRO-MyTestStrategy-003, account_id=InteractiveBrokers-DU1234567, opening_order_id=O-20230302-001-003-1, closing_order_id=None, entry=BUY, side=LONG, net_qty=99998.0, quantity=99_998.0, peak_qty=99_998.0, currency=JPY, avg_px_open=136.474, avg_px_close=0.0, realized_return=0.00000, realized_pnl=0 JPY, unrealized_pnl=0 JPY, ts_opened=1677750900000000000, ts_last=1677750900000000000, ts_closed=0, duration_ns=0).

```

As above was `TrailingStop` order so when that is Filled, here are logs. Now the `net_position=2.0`.
```
2023-03-02T09:58:11.683839400Z [INF] TESTER-001.InteractiveBrokersClient-002: ANSWER orderStatus {'order_id': 465, 'status': 'Filled', 'filled': Decimal('100000'), 'remaining': Decimal('0'), 'avg_fill_price': 136.508, 'perm_id': 1096798734, 'parent_id': 464, 'last_fill_price': 136.508, 'client_id': 2, 'why_held': '', 'mkt_cap_price': 0.0}
2023-03-02T09:58:11.686940200Z [INF] TESTER-001.InteractiveBrokersClient-002: ANSWER position {'account': 'DU1234567', 'contract': 2135093553296: 15016059,USD,CASH,,0,,,,,JPY,USD.JPY,USD.JPY,False,,,,combo:, 'position': Decimal('0'), 'avg_cost': 0.0}
2023-03-02T09:58:11.688429700Z [INF] TESTER-001.MyTestStrategy-003: <--[EVT] OrderFilled(instrument_id=USD/JPY.IDEALPRO, client_order_id=O-20230302-001-003-2, venue_order_id=465, account_id=InteractiveBrokers-DU1234567, trade_id=0000e215.63ffe256.01.01, position_id=USD/JPY.IDEALPRO-MyTestStrategy-003, order_side=SELL, order_type=TRAILING_STOP_MARKET, last_qty=100_000.0, last_px=136.508 JPY, commission=2.00 USD, liquidity_side=NO_LIQUIDITY_SIDE, ts_event=1677751091000000000).
2023-03-02T09:58:11.700014300Z [INF] TESTER-001.Portfolio: USD/JPY.IDEALPRO net_position=2.0
2023-03-02T09:58:11.700046900Z [INF] TESTER-001.MyTestStrategy-003: <--[EVT] PositionChanged(instrument_id=USD/JPY.IDEALPRO, position_id=USD/JPY.IDEALPRO-MyTestStrategy-003, account_id=InteractiveBrokers-DU1234567, opening_order_id=O-20230302-001-003-1, closing_order_id=None, entry=BUY, side=LONG, net_qty=2.0, quantity=2.0, peak_qty=99_998.0, currency=JPY, avg_px_open=136.474, avg_px_close=136.508, realized_return=0.00025, realized_pnl=3_400 JPY, unrealized_pnl=0 JPY, ts_opened=1677750900000000000, ts_last=1677751091000000000, ts_closed=0, duration_ns=0).
```

### Expected Behavior
Should get correct `net_position=100_000` in this case.

### Actual Behavior
Gives `net_position=998_980` in when opening and `net_position=2` when the same `100_000` is closed.

### Steps to Reproduce the Problem

1. Try placing order with USD/JPY

### Specifications

- OS platform: Win10
- Python version: 3.10.6
- `nautilus_trader` version: 1.170.0 (develop)

**Comment by cjdsellers** [comment]
I've figured out what causes this one, its when the commission currency == the base currency / underlying of the asset being traded, then its added onto the `last_qty`.

I actually don't think this is the correct implementation because `last_qty` won't match what is emitted in events from the exchange, and deduction of commission will already be accounted for in an account update event anyway.

I'll amend this behavior and see if any tests break.

**Comment by cjdsellers** [comment]
This could also be the cause of some anomalies in reconciliation, which I'll be getting to shortly.

**Comment by cjdsellers** [comment]
Should now be fixed from 8d3ebb766c0a1490c6eaa0e88275af1e536966ee on `develop`.

**Comment by rsmb7z** [comment]
I will test and update once the trading opens to include commissions scneario as don't have workaround to test using fixed commissions in backtest.
